rust/perspective-viewer/src/rust/session/metadata.rs

```diff
@@ -186,6 +187,23 @@ impl SessionMetadata {
         is_expr.unwrap_or_default()
     }
 
+    pub fn get_column_locator(&self, name: Option<String>) -> Option<ColumnLocator> {
+        if let Some(name) = name {
+            self.as_ref().and_then(|meta| {
+                if self.is_column_expression(&name) {
+                    Some(ColumnLocator::Expr(Some(name)))
+                } else {
+                    meta.column_names
+                        .iter()
+                        .find_map(|n| (n == &name).then_some(ColumnLocator::Plain(name.clone())))
+                }
+            })
+        } else {
+            // new expr
+            Some(ColumnLocator::Expr(None))
```

Comment:
Under what conditions does this function return `None`? I think this is a bit over-wrapped.

Reply:
This should return None when there is not a currently existing column. I removed the Expr(None) case.